### PR TITLE
parser: add new agg function `APPROX_COUNT_DISTINCT`

### PR DESCRIPTION
**Pull request by solotzg**
Signed-off-by: Tong Zhigao <tongzhigao@pingcap.com>

### What problem does this PR solve? 

- https://github.com/pingcap/tidb/issues/14632

### What is changed and how it works?

- Add new agg function `APPROX_COUNT_DISTINCT` to compute approximate result like count distinct.

- https://github.com/pingcap/tidb/pull/17175

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: ast/functions.go [hunk 1]
```diff
@@ -687,6 +687,8 @@ const (
 	AggFuncStddevSamp = "stddev_samp"
 	// AggFuncJsonObjectAgg is the name of json_objectagg function
 	AggFuncJsonObjectAgg = "json_objectagg"
+	// AggFuncApproxUniq is the name of approx_uniq function.
+	AggFuncApproxUniq = "approx_uniq"
```
Comment (zz-jason) [inline ast/functions.go:691]: what's the function name used in other databases?

**Comment by zz-jason** [comment]
@solotzg It's better to add a proposal firstly.

File: ast/functions.go (hunk 1 above)
Comment (solotzg) [reply ast/functions.go:691]: clickhouse use the name `uniq` for the same function and use `uniqExact` for count distinct.

Comment (zz-jason) [reply ast/functions.go:691]: how about hive, spark, impala?

Comment (solotzg) [reply ast/functions.go:691]: @ilovesoup

Comment (ilovesoup) [reply ast/functions.go:691]: presto: approx_distinct
spark: approx_count_distinct
impala: no special function but has a switch named appx_count_distinct
I suggest we named like approx_distinct for fewer keystrokes.

Comment (kennytm) [reply ast/functions.go:691]: MS SQL Server: [APPROX_COUNT_DISTINCT](https://docs.microsoft.com/en-us/sql/t-sql/functions/approx-count-distinct-transact-sql)

Oracle: [APPROX_COUNT_DISTINCT](https://docs.oracle.com/en/database/oracle/oracle-database/20/sqlrf/APPROX_COUNT_DISTINCT.html#GUID-50055A05-0187-4481-AFE5-2414F7227713)

BigQuery: [APPROX_COUNT_DISTINCT](https://cloud.google.com/bigquery/docs/reference/standard-sql/approximate_aggregate_functions)

Comment (solotzg) [reply ast/functions.go]: make function name: `APPROX_COUNT_DISTINCT` just like oracle and microsoft.

Comment (solotzg) [reply ast/functions.go]: @zz-jason @ilovesoup @lzmhhh123 PTAL

**Pull request merged**